internal/controller/imagerepository_controller_test.go

```diff
@@ -582,8 +594,13 @@ var _ = Describe("Image repository controller", func() {
 			assertSecretsGoneFromServiceAccounts()
 		})
 
-		It("should provision image repository for component, with update component annotation", func() {
-			assertProvisionRepository(true, false)
+		It("should provision image repository for component, with update component annotation, update also namespace secret", func() {
+			namespaceSecretKey := types.NamespacedName{Name: getNamespacePullSecretName(defaultNamespace), Namespace: defaultNamespace}
+			namespaceSecretEmptyData := generateDockerConfigJson("", "", "")
+			// namespace secret will exist also for following tests
```

Comment:
I think it's better to ensure the secret created in the following tests, so it's possible to run a test case independently.